docs/tasks/access-kubernetes-api/extend-api-custom-resource-definitions.md

```diff
@@ -0,0 +1,158 @@
+  group: stable.example.com
+  # version name to use for REST API: /apis/<group>/<version>
+  version: v1
+  # either Namespaced or Cluster
+  scope: Namespaced
+  names:
+    # plural name to be used in the URL: /apis/<group>/<version>/<plural>
+    plural: crontabs
+    # singular name to be used as alias on the CLI and for display
+    singular: crontab
+    # kind is normally the CamelCased singular type. Your resource manifests use this
+    kind: CronTab
+    # shortNames allow shorter string to match your resource on the CLI
+    shortNames:
+    - ct
+```
+
+And create it:
+
+```shell
+$ kubectl create -f resourcedefinition.yaml
+customresourcedefinitions "crontabs.stable.example.com" created
+```
+
+Then a new RESTful API endpoint is created at:
+
+`/apis/stable.example.com/v1/namespaces/<namespace>/crontabs/...`
+
+This endpoint URL can then be used to create and manage custom objects.
+The `kind` of these objects will be `CronTab` from the spec of the
+`CustomResourceDefinition` object we created above.
+
+
+## Creating Custom Objects
+
+After the `CustomResourceDefinition` object has been created you can create
+custom objects. Custom objects can contain custom fields. These fields can
+contain arbitrary JSON. 
+In the following example, a `cronSpec` and `image` custom fields are set to the
+custom object of kind `CronTab`.  The kind `CronTab` comes from the spec of the
+`CustomResourceDefinition` object we created above.
+
+If you save the following YAML to `my-crontab.yaml`:
+
+```yaml
+apiVersion: "stable.example.com/v1"
+kind: CronTab
+metadata:
+  name: my-new-cron-object
+cronSpec: "* * * * /5"
+image: my-awesome-cron-image
+```
+
+and create it:
+
+```shell
+$ kubectl create -f my-crontab.yaml
+crontab "my-new-cron-object" created
+```
+
+You can then manage our `CronTab` objects using kubectl. Note that resource
+names are not case-sensitive when using kubectl:
+
+```shell
+$ kubectl get crontab
+NAME                 KIND
+my-new-cron-object   CronTab.v1.stable.example.com
+```
+
+You can also view the raw JSON data. Here you can see that it contains the custom `cronSpec` and `image` fields from the yaml you used to create it:
+
+```yaml
+$ kubectl get ct -o yaml
+apiVersion: v1
+items:
+- apiVersion: stable.example.com/v1
+  cronSpec: '* * * * /5'
+  image: my-awesome-cron-image
+  kind: CronTab
+  metadata:
+    clusterName: ""
+    creationTimestamp: 2017-05-31T12:56:35Z
+    deletionGracePeriodSeconds: null
+    deletionTimestamp: null
+    name: my-new-cron-object
+    namespace: default
+    resourceVersion: "285"
+    selfLink: /apis/stable.example.com/v1/namespaces/default/crontabs/my-new-cron-object
+    uid: 9423255b-4600-11e7-af6a-28d2447dc82b
+kind: List
+metadata:
+  resourceVersion: ""
+  selfLink: ""
+```
+
+## Advanced Topics
+### Finalizers
+CustomResources (objects created in the schema defined by CustomResourceDefintions)
+support finalizers.  If you add a `metadata.finalizers` stanza like
+
+```yaml
```

Comment:
I would add `apiVersion`/`kind` for `CronTab` to make it clear by example which type of object we're talking about.